packages/picasso/src/utils/useBoolean/use-boolean.ts

```diff
@@ -0,0 +1,30 @@
+import { useState } from 'react'
+
+/** Set value to true */
+type SetTruthy = () => void
+/** Set value to false */
+type SetFalsy = () => void
+/** Toggle the value */
+type Toggle = () => void
+type IsTruthy = boolean
+
+type UseBooleanType = (
+  defaultValue?: boolean
+) => [IsTruthy, SetTruthy, SetFalsy, Toggle]
+
+/**
+ *
+ * @param defaultValue?: Boolean
+ * @returns [isTruthy, setTruthy(), setFalsy(), toggle()]
+ */
+const useBoolean: UseBooleanType = (defaultValue = false) => {
+  const [isTruthy, setBoolean] = useState(defaultValue)
+
+  const setTruthy: SetTruthy = () => setBoolean(true)
+  const setFalsy: SetFalsy = () => setBoolean(false)
+  const toggle: Toggle = () => setBoolean(value => !value)
```

Comment:
We could use `useCallback` here, so when someone spams any of the state setters they won't be evaluated. Just a nit, not a big deal I think
```suggestion
  const setTruthy: SetTruthy = useCallback(() => setBoolean(true), [])
  const setFalsy: SetFalsy = useCallback(() => setBoolean(false), [])
  const toggle: Toggle = useCallback(() => setBoolean(value => !value), [])
```

Reply:
@LashaJini I believe react covers this for us, since `setBoolean` is fn from `useState`

Reply:
I am against it @LashaJini. Here you may memoize for nothing. Users of this hook can decide what to memoize.

Generally, you memoize what you are passing to your child. Memoizing inside a hook may not be that beneficial.

Reply:
I see your point and agree :+1: